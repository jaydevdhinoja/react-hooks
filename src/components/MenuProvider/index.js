import React, { createContext, useContext, useState } from 'react';
import Items from '../../items';

const defaultState = {
	menuItems: Items,
	menuPreviewItems: []
};

export const MenuItemsContext = createContext(defaultState);

const MenuItemsProvider =  ({children}) => {

    const InitialState = () => defaultState;
    const [ initialState, setInitialState ] = useState(InitialState());
    // const removeMenuPreviewItem = (item) => {
    //     const menuPreviewItems = initialState.menuPreviewItems.filter(menuItem => menuItem.id !== item.id)
    //     setInitialState({ ...initialState, menuPreviewItems});
    // }
    return (
		<MenuItemsContext.Provider
			value={{
				...initialState,
				addInMenuPreview: (item) => {
                    debugger
                    const updatedMenuItems = initialState.menuItems.filter(menuItem => menuItem.id !== item.id)
					setInitialState({ ...initialState, menuItems: updatedMenuItems, menuPreviewItems: [...initialState.menuPreviewItems, item] });
				},
				removeMenuPreviewItem : (item) => {
                    const menuPreviewItems = initialState.menuPreviewItems.filter(menuItem => menuItem.id !== item.id)
                    const updatedMenuItems = [...initialState.menuItems, item];
                    setInitialState({ ...initialState, menuItems: updatedMenuItems, menuPreviewItems});
                }
			}}
		>
			{children}
		</MenuItemsContext.Provider>
	);
};

export default MenuItemsProvider;

export const useMenuItemsContext = () => {
	return useContext(MenuItemsContext);
};