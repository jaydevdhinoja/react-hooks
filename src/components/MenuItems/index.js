import React from 'react';
import {useMenuItemsContext} from '../MenuProvider';

const MenuItems = () => {
    const { menuItems, addInMenuPreview } = useMenuItemsContext();
    const items = menuItems.map( item => 
        <li key={item.id} className="item" onClick={() => addInMenuPreview(item)} >
        <h2>{item.name}</h2>
        <p>
          {
            item.dietaries.map(dietary => <span key={dietary} className="dietary">{dietary}</span>)
          }  
        </p>
      </li>
    );
    
    return (
        <ul className="item-picker">
            {items}
        </ul>
    )
}

export default MenuItems;