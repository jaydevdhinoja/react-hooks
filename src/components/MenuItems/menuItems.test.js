import React from "react";
import MenuItems from "./index";
import {MenuItemsContext} from '../MenuProvider';
import Items from '../../items';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

const defaultState = {
	menuItems: Items,
    menuPreviewItems: [],
    addInMenuPreview: jest.fn()
};

function renderWithContext(node, { value, ...options }) {
	return render(<MenuItemsContext.Provider value={value}>{node}</MenuItemsContext.Provider>, options);
}

describe("Menu Provider context snapshot", () => {
  it("should render children for snapshot", () => {
    const tree = renderWithContext(<MenuItems/>, {value: defaultState});
    expect(tree).toMatchSnapshot();
  });

  it("renders menu items", () => {
    const { getByText } = renderWithContext(<MenuItems/>, {value: defaultState});
    expect(getByText(/Hake & Smoky Chickpeas, Brown Rice & Quinoa, Roasted Roots/i)).toBeInTheDocument();
  });

  it("should be able to click and add the item in preview", () => {
    const { getByText } = renderWithContext(<MenuItems/>, {value: defaultState});
    const link = getByText(/Hake & Smoky Chickpeas, Brown Rice & Quinoa, Roasted Roots/i);
    fireEvent.click(link);
  });
});