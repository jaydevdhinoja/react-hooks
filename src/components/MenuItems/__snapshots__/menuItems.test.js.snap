// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Provider context snapshot should render children for snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="item-picker"
      >
        <li
          class="item"
        >
          <h2>
            Kale Caesar Pasta, Turmeric Satay Broccoli & Lemon Cashew Greens
          </h2>
          <p>
            <span
              class="dietary"
            >
              v
            </span>
            <span
              class="dietary"
            >
              ve
            </span>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              n!
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Hake & Smoky Chickpeas, Brown Rice & Quinoa, Roasted Roots
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              rsf
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Dill & Swiss Chard Potato Cakes, Summer Tabbouleh & Roasted Roots
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              v
            </span>
            <span
              class="dietary"
            >
              ve
            </span>
            <span
              class="dietary"
            >
              n!
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Hake & Smoky Chickpeas, Herby Potatoes & Turmeric Satay Broccoli
          </h2>
          <p>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              rsf
            </span>
            <span
              class="dietary"
            >
              n!
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Cuban Beef, Brown Rice & Quinoa, Green Pepper & Butterbean Salad
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Thai Chicken, Brown Rice & Quinoa, Lemon & Cashew Greens
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              rsf
            </span>
            <span
              class="dietary"
            >
              n!
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Cuban Beef, Herby Potatoes & Turmeric Satay Broccoli
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              n!
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Thai Chicken, Brown Rice & Quinoa & Turmeric Satay Broccolli
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              rsf
            </span>
            <span
              class="dietary"
            >
              n!
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Mackerel Salad Box
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Pasta Salad Box
          </h2>
          <p>
            <span
              class="dietary"
            >
              ve
            </span>
            <span
              class="dietary"
            >
              v
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Tuna Salad Box
          </h2>
          <p />
        </li>
        <li
          class="item"
        >
          <h2>
            Coconut Blondie with Dark Chocolate
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              v
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Cinnamon Swirl Muffin
          </h2>
          <p>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              df
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Apple, Date and Walnut Muffin
          </h2>
          <p>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
            <span
              class="dietary"
            >
              n!
            </span>
            <span
              class="dietary"
            >
              v
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Blueberry and Banana Muffin
          </h2>
          <p>
            <span
              class="dietary"
            >
              v
            </span>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Ombar Chocolate Bar - Dark 72%
          </h2>
          <p>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              ve
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Ombar Bar - Milk
          </h2>
          <p>
            <span
              class="dietary"
            >
              ve
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Dark Chocolate Brownie
          </h2>
          <p>
            <span
              class="dietary"
            >
              v
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Mangajo Pomegranate
          </h2>
          <p>
            <span
              class="dietary"
            >
              ve
            </span>
            <span
              class="dietary"
            >
              df
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
          </p>
        </li>
        <li
          class="item"
        >
          <h2>
            Mangajo Acai
          </h2>
          <p>
            <span
              class="dietary"
            >
              ve
            </span>
            <span
              class="dietary"
            >
              v
            </span>
            <span
              class="dietary"
            >
              gf
            </span>
          </p>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="item-picker"
    >
      <li
        class="item"
      >
        <h2>
          Kale Caesar Pasta, Turmeric Satay Broccoli & Lemon Cashew Greens
        </h2>
        <p>
          <span
            class="dietary"
          >
            v
          </span>
          <span
            class="dietary"
          >
            ve
          </span>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            n!
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Hake & Smoky Chickpeas, Brown Rice & Quinoa, Roasted Roots
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            rsf
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Dill & Swiss Chard Potato Cakes, Summer Tabbouleh & Roasted Roots
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            v
          </span>
          <span
            class="dietary"
          >
            ve
          </span>
          <span
            class="dietary"
          >
            n!
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Hake & Smoky Chickpeas, Herby Potatoes & Turmeric Satay Broccoli
        </h2>
        <p>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            rsf
          </span>
          <span
            class="dietary"
          >
            n!
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Cuban Beef, Brown Rice & Quinoa, Green Pepper & Butterbean Salad
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Thai Chicken, Brown Rice & Quinoa, Lemon & Cashew Greens
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            rsf
          </span>
          <span
            class="dietary"
          >
            n!
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Cuban Beef, Herby Potatoes & Turmeric Satay Broccoli
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            n!
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Thai Chicken, Brown Rice & Quinoa & Turmeric Satay Broccolli
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            rsf
          </span>
          <span
            class="dietary"
          >
            n!
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Mackerel Salad Box
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Pasta Salad Box
        </h2>
        <p>
          <span
            class="dietary"
          >
            ve
          </span>
          <span
            class="dietary"
          >
            v
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Tuna Salad Box
        </h2>
        <p />
      </li>
      <li
        class="item"
      >
        <h2>
          Coconut Blondie with Dark Chocolate
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            v
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Cinnamon Swirl Muffin
        </h2>
        <p>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            df
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Apple, Date and Walnut Muffin
        </h2>
        <p>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
          <span
            class="dietary"
          >
            n!
          </span>
          <span
            class="dietary"
          >
            v
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Blueberry and Banana Muffin
        </h2>
        <p>
          <span
            class="dietary"
          >
            v
          </span>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Ombar Chocolate Bar - Dark 72%
        </h2>
        <p>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            ve
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Ombar Bar - Milk
        </h2>
        <p>
          <span
            class="dietary"
          >
            ve
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Dark Chocolate Brownie
        </h2>
        <p>
          <span
            class="dietary"
          >
            v
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Mangajo Pomegranate
        </h2>
        <p>
          <span
            class="dietary"
          >
            ve
          </span>
          <span
            class="dietary"
          >
            df
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
        </p>
      </li>
      <li
        class="item"
      >
        <h2>
          Mangajo Acai
        </h2>
        <p>
          <span
            class="dietary"
          >
            ve
          </span>
          <span
            class="dietary"
          >
            v
          </span>
          <span
            class="dietary"
          >
            gf
          </span>
        </p>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
